source/client-side-encryption/client-side-encryption.rst

```diff
@@ -453,45 +458,99 @@ the following as a template:
 
 See `What's the deal with metadataClient, keyVaultClient, and the internal client?`_
 
+.. _AWSKMSOptions:
+.. _KMSProvidersMap:
+
 kmsProviders
 ^^^^^^^^^^^^
-Multiple KMS providers may be specified. The kmsProviders map values differ by
-provider ("aws", "azure", "gcp", "local", and "kmip"). The "local" provider is
-configured with master key material. The external providers are configured with
-credentials to authenticate.
+
+The ``kmsProviders`` property may be specified on ClientEncryptionOpts_ or
+AutoEncryptionOpts_. Multiple KMS providers may be specified, each using a
+specific property on the KMSProvidersMap_ object. The options for each type of
+provider differ by the provider name ("aws", "azure", "gcp", "local", and
+"kmip"). The "local" provider is configured with master key material. The
+external providers are configured with credentials to authenticate.
 
 .. code:: typescript
 
-   aws: {
-      accessKeyId: String,
-      secretAccessKey: String,
-      sessionToken: Optional<String> // Required for temporary AWS credentials.
-   }
+   interface KMSProvidersMap {
```

Comment:
I support the addition of the interfaces. Can you extract the changes unrelated to the on-demand credentials into. separate PR? That can be reviewed and merged independently of DRIVERS-2280, and I think will help clarify what has changed in the spec when other implementers review the diff.